arxlive_spotlight_annotation/src/node_modules/dbpedia/spotlight.mjs

```diff
@@ -26,21 +65,144 @@ export const annotate = async (
+					percentageOfSecondRank: parseFloat(result['@percentageOfSecondRank']),
 				};
 		  })
 		: [];
 
 	return {
-		confidence: results['@confidence'],
+		confidence: spotlightAnnotation['@confidence'],
 		results: resultSubset,
 	};
 };
+
+/**
+ * The final form of resource, this is the same as {@link ReducedResource}, however
+ * the confidence property has been added to the objects values.
+ * All properties below are the exact same.
+ * @typedef DBpediaEntity
+ * @type {Object}
+ * @property {string} URI - the Unique Resource Identifier for this resource.
+ * @property {string} surfaceForm - the original string used to produce this resource.
+ * @property {number} similarityScore - cosine similarity between the context vectors and the context surrounding the surface form.
+ * @property {number} percentageOfSecondRank - the relative difference in topic score between the first and the second ranked resource.
+ * @property {number} confidence - the highest confidence at which this resource could be found. This means that all annotations performed at a lower confidence than the one given here will also produce this annotation.
+ */
+
+/**
+ * @function reduceAnnotationResults 
+ * @description Maps {@link ParsedAnnotation} objects using their confidedence.
+ * @param {Object.<string, ParsedAnnotation>} parsedAnnotationByConfidence - an object where {@link ReducedResource} objects are mapped by the condience with which they were produced.
+ * @returns {DBpediaEntity[]} - a list of annotated entities.
+ */
+export const reduceAnnotationResults = (parsedAnnotationByConfidence) => {
+	const seen = new Set();
+	const confidenceKeys = [
+		'1.0',
+		'0.9',
+		'0.8',
+		'0.7',
+		'0.6',
+		'0.5',
+		'0.4',
+		'0.3',
+		'0.2',
+		'0.1',
+		'0.0',
+	];
+	const reducedTerms = confidenceKeys.reduce((prev, confidence) => {
+		const terms = parsedAnnotationByConfidence[confidence];
+		const unseenTerms = terms.filter((term) => !seen.has(term.URI));
+		return prev.concat(
+			unseenTerms.map((term) => {
+				seen.add(term.URI);
+				return { ...term, confidence: parseFloat(confidence) };
+			})
+		);
+	}, []);
+	return reducedTerms;
+};
+
+/**
+ * @function annotateDocument
+ * @description takes an Elastic search document from Arxlive and annotates the abstract_article field.
+ * @param {Object} doc - an ElasticSearch document from the Arxlive domain. 
+ * @param {string} endpoint - the endpoint pointing to the Spotlight REST API.
+ * @returns {DBpediaEntity[]} a list of annotated entities.
+ */
+export const annotateDocument = async (doc, endpoint = spotlightEndpoint) => {
+	const spotLightPromises = confidenceValues.map((confidence) => {
+		return annotate(doc._source.textBody_abstract_article, confidence, {
+			endpoint,
+		});
+	});
+	/** @type {SpotlightAnnotation[]} */
+		const spotlightResults = await Promise.all(spotLightPromises);
+	/** @type {ParsedAnnotation[]} */
+	const spotlightTerms = spotlightResults.map(parseAnnotationResults);
```

Comment:
Indentation of the first `const` seems off; also I'd use a blank line above comments for readability, like:
```suggestion

	/** @type {SpotlightAnnotation[]} */
	const spotlightResults = await Promise.all(spotLightPromises);
	
	/** @type {ParsedAnnotation[]} */
	const spotlightTerms = spotlightResults.map(parseAnnotationResults);
```